packages/main/src/plugin/filesystem-monitoring.ts

```diff
@@ -58,6 +60,11 @@ export class FileSystemWatcherImpl implements containerDesktopAPI.FileSystemWatc
   readonly onDidDelete: containerDesktopAPI.Event<containerDesktopAPI.Uri> = this._onDidDelete.event;
 
   dispose(): void {
+    if (this.watcher) {
+      this.watcher.close().catch((err: unknown) => {
```

Comment:
I'm not sure if we want to do this, or if we can change the signature of the method to `async dispose(): Promise<void>`. I think that it won't respect the contract of the `Disposable` class anymore if we do this.

Reply:
we can't change the signature